lib/utils/addr/addr.go

```diff
@@ -355,3 +355,15 @@ func ReplaceUnspecifiedHost(addr *NetAddr, defaultPort int) string {
 	port := addr.Port(defaultPort)
 	return net.JoinHostPort("localhost", strconv.Itoa(port))
 }
+
+// ClientIPFromConn extracts host from provided remote address.
+func ClientIPFromConn(conn net.Conn) (string, error) {
```

Comment:
Move func to its own file? It seems disjointed from the rest.

Reply:
It seemed related in the sense that it's IP parsing.  I can put it in a separate file, but I felt it was related enough to combine.  Let me know your preference